addons/website_sale_comparison/static/src/interactions/cookies_bar.js

```diff
@@ -0,0 +1,77 @@
+import { CookiesBar } from "@website/interactions/cookies/cookies_bar";
+import { patch } from "@web/core/utils/patch";
+
+patch(CookiesBar.prototype, {
+    /**
+     * @override
+     */
+    setup() {
+        super.setup();
+        // Listen for the custom event to handle the overlap of the compare
+        // button with the cookies bar modal.
+        this.el.addEventListener("cookies_bar_shown", () => {
+            this.handleCompareOverlap();
+        });
+    },
+    /**
+     * @override
+     */
+    showPopup() {
+        super.showPopup();
+        // Initially when the modal is shown, its height is 0, so we need to
+        // wait for the modal to be fully rendered before adjusting the position
+        // of the compare button. Through the `--move-cookie-over-modal` CSS
+        // variable, we can dynamically adjust the position of elements based on
+        // modal changes.
+        this.waitForTimeout(() => {
+            this.handleCompareOverlap();
+        }, 0);
+    },
+    /**
+     * @override
+     *
+     * Remove the custom css for `--move-cookie-over-modal`, to position the
+     * compare button at the bottom.
+     */
+    onHideModal() {
+        super.onHideModal();
+        const productPanelEl = document.querySelector("[name='product_comparison_button']");
+        productPanelEl?.style.removeProperty("--move-cookie-over-modal");
+    },
+    /**
+     * Resolved the issue where the compare button was hidden due to the modal
+     * appearing over it. Added CSS `--move-cookie-over-modal` to dynamically
+     * adjust the position of elements based on modal changes.
+     */
+    handleCompareOverlap() {
+        const productCompareButtonEl = document.querySelector("[name='product_comparison_button']");
+        if (!productCompareButtonEl) {
+            return;
+        }
+
+        const cookieModelEl = this.el.querySelector(".modal");
+        if (
+            !cookieModelEl ||
+            cookieModelEl.style.display === "none" ||
+            cookieModelEl.classList.contains("o_cookies_popup") ||
+            !cookieModelEl.classList.contains("s_popup_no_backdrop")
+        ) {
+            return;
+        }
+        const cookieModalDialogEl = cookieModelEl.querySelector(".modal-dialog");
+        const isCookiebarLarge = ["s_popup_size_full", "modal-xl"].some((cls) =>
+            cookieModalDialogEl.classList.contains(cls)
+        );
+
+        if (isCookiebarLarge) {
+            const bottom = cookieModalDialogEl.querySelector(".modal-content").offsetHeight
+                ? `${cookieModalDialogEl.querySelector(".modal-content").offsetHeight}px`
+                : "";
+            const addSpacer = getComputedStyle(cookieModalDialogEl).paddingBottom;
```

Comment:
`addSpacer` ? it would be much better is i simply name it `padding` as we are getting that only as a value.